src/SapAct/Extensions/JsonElementExtensions.cs

```diff
@@ -61,7 +62,8 @@ public static Dictionary<string, string> ExportToFlattenedDictionary(this JsonEl
 		{
 			foreach (var field in dataField.EnumerateObject())
 			{
-				dataFields.Add(field.Name, field.Value.ToString());
+				if (!dataFields.ContainsKey(field.Name))
```

Comment:
Just a thought - if you extract the data fields _first_, then enumerate over the top-level fields, they'll override the data fields if you use `dataFields[key] = value;` rather than `.Add(key, value)`? Saves having to do a bunch of contains checks 😃

Reply:
yeah that's not a bad approach... we will somewhat hide the intention of "merging" but it should still be clear enough I think

Reply:
Can fix that with a brief comment in the top-level loop noting that top-level properties are explicitly and purposefully overriding any properties extracted from the data object?

Reply:
agreed